tests/integ/modin/test_faster_pandas.py

```diff
@@ -1673,6 +1673,40 @@ def test_series_to_datetime(session):
         assert_series_equal(snow_result, native_result)
 
 
+@sql_count_checker(query_count=3)
+def test_to_snowpark(session):
+    with session_parameter_override(
+        session, "dummy_row_pos_optimization_enabled", True
+    ):
+        # create tables
+        table_name = Utils.random_name_for_temp_object(TempObjectType.TABLE)
+        session.create_dataframe(
+            native_pd.DataFrame([[2, 12], [1, 11], [3, 13]], columns=["A", "B"])
+        ).write.save_as_table(table_name, table_type="temp")
+
+        # create snow dataframes
+        df = pd.read_snowflake(table_name)
+        sdf = df.to_snowpark(index=False)
+        snow_result = sdf.to_snowpark_pandas()
+
+        # verify that the input dataframe has a populated relaxed query compiler
+        assert df._query_compiler._relaxed_query_compiler is not None
+        assert df._query_compiler._relaxed_query_compiler._dummy_row_pos_mode is True
+        # verify that the output dataframe also has a populated relaxed query compiler
+        assert snow_result._query_compiler._relaxed_query_compiler is not None
+        assert (
+            snow_result._query_compiler._relaxed_query_compiler._dummy_row_pos_mode
+            is True
+        )
+
+        # create pandas dataframes
+        native_df = df.to_pandas()
+        native_result = native_df
+
+        # compare results
+        assert_frame_equal(snow_result, native_result)
```

Comment:
```suggestion
        # compare results
        assert_frame_equal(snow_result, df.to_pandas())
```

Same q as for the to_snowflake PR, is this check deterministic?

Reply:
Also added an explicit sort to ensure determinism.